gammapy/makers/map.py

```diff
@@ -319,10 +325,15 @@ def make_psf(self, geom, observation):
         """
         psf = observation.psf
 
-        if isinstance(psf, RecoPSFMap):
-            return RecoPSFMap(psf.psf_map.interp_to_geom(geom))
-        elif isinstance(psf, PSFMap):
-            return PSFMap(psf.psf_map.interp_to_geom(geom))
+        if isinstance(psf, PSFMap):
+            if psf.exposure_map:
+                exposure_map = psf.exposure_map.interp_to_geom(
+                    geom.squash(axis_name="rad")
+                )
+            else:
+                exposure_map = None
+            return psf.__class__(psf.psf_map.interp_to_geom(geom), exposure_map)
```

Comment:
If exposure_map is None, the interp would return all nans, no?

Reply:
`exposure_map` can be None bur not `psf_map`